src/ipc-handlers.ts

```diff
@@ -524,7 +524,7 @@ export async function openLocalPath( _event: IpcMainInvokeEvent, path: string )
 export async function getThemeDetails( event: IpcMainInvokeEvent, id: string ) {
 	const server = SiteServer.get( id );
 	if ( ! server ) {
-		throw new Error( 'Site not found.' );
+		return null;
```

Comment:
I think this change is not related to this PR.

Reply:
I got an error at some point, before the refactor, when I tried different things to make the previous code work, so I decided to leave it. I'll move it back, as it was before and re test

Reply:
It seems, it's not needed anymore, so I removed it. I guess it has crashing the app due to race conditions.